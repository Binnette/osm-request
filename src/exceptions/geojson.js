export function GeoJSONException(message) {
  this.message = message;
  this.name = 'GeoJSONException';
}
