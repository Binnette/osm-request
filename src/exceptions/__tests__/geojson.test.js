import { GeoJSONException } from '../geojson';

describe('GeoJSON exceptions', () => {
  describe('GeoJSONException', () => {
    it('Should be well formatted', () => {
      const message = 'This is a test';
      const myFunction = () => {
        throw new GeoJSONException(message);
      };

      expect(myFunction).toThrow(GeoJSONException);
      expect(myFunction).toThrow(message);
      expect(myFunction).toThrowErrorMatchingSnapshot();
    });
  });
});
