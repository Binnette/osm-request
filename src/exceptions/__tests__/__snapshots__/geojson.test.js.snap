// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoJSON exceptions GeoJSONException Should be well formatted 1`] = `"This is a test"`;
